api/provider/aws/environment_scaler.go

```diff
@@ -1,14 +1,758 @@
+}
+
+func (e *EnvironmentScaler) getResourceConsumers_PendingServices(clusterName string) ([]ResourceConsumer, error) {
+	var resourceConsumers []ResourceConsumer
+
+	listServicesInput := &ecs.ListServicesInput{}
+	listServicesInput.SetCluster(clusterName)
+
+	serviceARNs := []*string{}
+	listServicesPagesFN := func(output *ecs.ListServicesOutput, lastPage bool) bool {
+		serviceARNs = append(serviceARNs, output.ServiceArns...)
+
+		return !lastPage
+	}
+
+	if err := e.Client.ECS.ListServicesPages(listServicesInput, listServicesPagesFN); err != nil {
+		return nil, err
+	}
+
+	services := []*ecs.Service{}
+	if len(serviceARNs) > 0 {
+		// The SDK states that you can specify up to 10 services in one DescribeServices operation:
+		// https://github.com/aws/aws-sdk-go/blob/ee1f179877b2daf2aaabf71fa900773bf8842253/service/ecs/api.go#L5420
+		// (aws-sdk-go version 1.12.19, as stated in layer0/Gopkg.toml)
+		for i := 0; i < len(serviceARNs); i += 10 {
+			end := i + 10
+
+			if end > len(serviceARNs) {
+				end = len(serviceARNs)
+			}
+
+			describeServicesInput := &ecs.DescribeServicesInput{}
+			describeServicesInput.SetCluster(clusterName)
+			describeServicesInput.SetServices(serviceARNs[i:end])
+
+			output, err := e.Client.ECS.DescribeServices(describeServicesInput)
+			if err != nil {
+				return nil, err
+			}
+
+			services = append(services, output.Services...)
+		}
+	}
+
+	for _, service := range services {
+		deployIDCopies := map[string]int64{}
+		for _, d := range service.Deployments {
+			desiredCount := aws.Int64Value(d.DesiredCount)
+			runningCount := aws.Int64Value(d.RunningCount)
+			pendingCount := aws.Int64Value(d.PendingCount)
+			if numPending := desiredCount - (runningCount + pendingCount); numPending > 0 {
+				deployIDCopies[aws.StringValue(d.Id)] = numPending
+			}
+		}
+
+		if len(deployIDCopies) == 0 {
+			continue
+		}
+
+		// iterate through deploys
+		for deployID := range deployIDCopies {
+			c, err := e.getContainerResourceFromDeploy(deployID)
+			if err != nil {
+				return nil, err
+			}
+
+			resourceConsumers = append(resourceConsumers, c...)
+		}
+	}
+
+	return resourceConsumers, nil
+}
+
+func (e *EnvironmentScaler) getResourceConsumers_TasksInECS(clusterName string) ([]ResourceConsumer, error) {
+	var (
+		taskARNs          []string
+		resourceConsumers []ResourceConsumer
+	)
+
+	fn := func(output *ecs.ListTasksOutput, lastPage bool) bool {
+
+		for _, taskARN := range output.TaskArns {
+			taskARNs = append(taskARNs, aws.StringValue(taskARN))
+		}
+
+		return !lastPage
+	}
+
+	startedBy := e.Config.Instance()
+	for _, status := range []string{ecs.DesiredStatusRunning, ecs.DesiredStatusStopped} {
+		input := &ecs.ListTasksInput{}
+		input.SetCluster(clusterName)
+		input.SetDesiredStatus(status)
+		input.SetStartedBy(startedBy)
+		if err := e.Client.ECS.ListTasksPages(input, fn); err != nil {
+			return nil, err
+		}
+	}
+
+	for _, taskARN := range taskARNs {
+		task, err := e.TaskProvider.Read(taskARN)
```

Comment:
is this a literal `provider.TaskProvider`? Because `TaskProvider.Read` takes a `TaskID`, not a `TaskARN`

Reply:
Yeah, that's right - if the removal of checking-ECS-for-pending-tasks is the correct thing to do, however, this point is moot.